astro_reduction_tool.py

```diff
@@ -0,0 +1,222 @@
+from astropy import units as u
+import os
+from astropy.nddata import CCDData
+import numpy as np
+from astropy.stats import mad_std
+import ccdproc as ccdp
+from astropy.io import fits
+import logging
+from datetime import datetime
+import argparse
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+console_handler = logging.StreamHandler()
+console_handler.setLevel(logging.INFO)
+logger.addHandler(console_handler)
+
+
+class ccd_data:
+    def __init__(self, data_list, name_list):
+        self.data_list = data_list
+        self.name_list = name_list
+
+    def __str__(self):
+        return "List 1: {}\nList 2: {}".format(self.data_list, self.name_list)
+
+def inv_median(a):
+    return 1 / np.median(a)
+
+def read_path_into_CCDData(data_path, keys, _filter=None, _ifc_mf = [] ):
+    data_list = []
+    name_list = []
+    if not os.path.isdir(
+        data_path
+    ):
+        logger.error(f"No directory {data_path}")
+        raise ValueError(f"No directory {data_path}")
+    if len(_ifc_mf) == 0:
+        ccd_collection = ccdp.ImageFileCollection(data_path, keywords=keys)
+        if _filter:
+            ccd_collection = ccd_collection.files_filtered(filter=_filter)
+        else:
+            ccd_collection = ccd_collection.files
+    else:
+        ccd_collection = _ifc_mf
+    if len(ccd_collection) != 0:
+        logger.info(f"Loading data from: {data_path}")
+        for ccd_path in ccd_collection:
+            dir = os.path.join(data_path, ccd_path)
+            ccd_file = CCDData(
+                fits.getdata(dir), meta=fits.getheader(dir, keywords=keys), unit="adu"
+            )
+            header_value = ccd_file.meta.get("combined")
+            if header_value is None or len(_ifc_mf) != 0:
+                data_list.append(ccd_file)
+                name_list.append(ccd_path)
+                now = datetime.now()
+                formatted_time = now.strftime("[%Y-%m-%dT%H:%M:%S.%f]")
+                logging.info(
+                    "%s  Loading %s file %s (1 of 1)",
+                    formatted_time,
+                    ccd_file.header["IMAGETYP"],
+                    ccd_path,
+                )
+    ccd_data_list = ccd_data(data_list,name_list)
+    return ccd_data_list
+
+def debias_dedark(file_list, master_dark, master_bias):
+    for h in range(0, len(file_list.data_list)):
+        ccd_file = file_list.data_list[h]
+        ccd_name = file_list.name_list[h]
```

Comment:
```suggestion
    for ccd_file_idx, ccd_file in enumerate(file_list.data_list):
        ccd_name = file_list.name_list[ccd_file_idx]
```

Zaleta `enumerate()` jest to ze daje nam obiekt z listy i jego indeks jednoczesnie, nie trzeba przy tym stosowac range + len.